### ISSUE TITLE
Rename this project and nuget package

### ISSUE THREAD

**Issue by dsyme** [colinbull/FSharpEnt#8]
Hi @colinbull ,

I'm doing an audit of F# nuget packages searching for "FSharp" and this one came up as one that uses a two-word package name.

Ideally we should rename this project (and more importantly the corresponding nuget package) to follow the [recently published guidelines](http://fsharp.github.io/2014/09/19/fsharp-libraries.html).

However because of the nature of the library it's a little bit tricky.Do you have any ideas on what would be most appropriate?  

Thanks!
Don

**Comment by colinbull** [comment]
Hi @dsyme,

Yes, I saw the conversation on twitter, and I have been thinking about this. In general this is probably a badly named library anyway as it has very little to actually do with the enterprise. Most of the modules in the core library are just extensions or wrappers around existing .NET framework classes and these types can probably be moved to FSharpx (if FSharpx is remaining of course). This has happened in the past with the various List, Seq, Array extensions and the RingBuffer data structures, and the HTML parser has been moved to FSharp.Data as part of the HtmlProvider work. However there are other modules that deal with Geometry problems, intersecting lines, interpolation etc. These will probably need moving to some other library which is a collection of general math functions, although I don't know an FSharp.\* library of this type. The Google data source implementation can probably move to Foogle.Charts and the RabbitMQ stuff probably needs a library of own, but it is a little incomplete as it was done just to satisfy my requirements at E.ON.

In summary, rather than renaming, I'm advocating splitting this library up and moving types to other libraries or creating more specific libraries as per the guidelines. 

@simontcousins probably has a view on this as he also use the library for projects. 

What do you think?

Cheers
Colin

**Comment by simoncousins** [comment]
I agree with Colin. The library doesn't have a focus, it was a dumping ground for things we found useful in developing our apps. Now that we have a rich landscape of more focused F# libraries it makes sense to split it up and retire it.

**Comment by dsyme** [comment]
I agree to some extent.  However, the FSharpx.\* set of packages is still too much a dumping ground for me to recommend it as the destination for these bits of functionality.  There has been a move to improve this, with the type providers being split out to and the (somewhat scattered) set of collections split out to [FSharpx.Collections](http://fsprojects.github.io/FSharpx.Collections/).  The core FSharpx project itself has problems
- it doesn't have up-to-date generated tutorials or docs (see http://fsprojects.github.io/fsharpx/)
- the parts that have been "split out" are still there 
- it has only had 3 commits in the last 3 months, so there's not a lot going on, but there are still 24 open issues

It's fine to gradually retire FSharp.Enterprise, but a much simpler alternative could just be to rename to some code name, like "FSharpEnterpriseHelpers", "FsEnterpriseHelpers", "BlinkyBlink.Helpers" or the like.  it's fine to have adhoc collections of helpers like this, we should just gradually move them out of the "FSharp.*" namespace.

Cheers,
Don

**Comment by colinbull** [comment]
O.K in that case then I'll move it back to it's original name, FSharpEnt. Then as new projects come about I'll move the stuff out that is applicable.
